pandas/_core/groupby/indexing.py

```diff
@@ -0,0 +1,304 @@
+from __future__ import annotations
```

Comment:
sometimes git understands when a file is just renamed.  not sure how that happens.  can we make it happen here?

Reply:
I think no if we want to do this all in one shot. I believe the issue is that we are renaming but then also recreating the files with different content, and git doesn't recognize this. If we were to split this up into two PRs: (1) rename core -> _core and then (2) add back the core files for backwards compatibility / deprecation, then I think git would recognize it.

Breaking it up into these two steps would certainly make review easier, but we'd need to make sure we carry it all out before the release.